tests/resources/helper_functions.py

```diff
@@ -384,16 +383,20 @@ def get_mock_provider_fees(mock_type, wallet, valid_until=0):
     }
 
 
-@contextlib.contextmanager
-def delay_transaction():
-    yield
-    time.sleep(2)
-
-
-def confirm_failed(tx, message):
-    chain_message = interrogate_blockchain_for_reverts(tx)
-    assert tx.status == 0
-    assert message in chain_message
+def confirm_failed(contract, fn_name, fn_args, message):
+    func = getattr(contract, fn_name)
+    timeout = time.time() + 10
+    while time.time() < timeout:
+        try:
+            tx = func(*fn_args)
+            time.sleep(1)
+            chain_message = interrogate_blockchain_for_reverts(tx)
+            if tx.status == 0 and message in chain_message:
+                break
+        except TypeError:
+            pass
```

Comment:
I would leave the TypeError nondescript for now, either way we can not assert it but only check if it is that one with "int base decimal". However, before the transaction is confirmed I've seen more possible TypeErrors, including "incompatible operand" ones so it would be a really hard thing to do right now to catalogue them all. Maybe we can create a follow-up issue for the specific TypeError instead? And work on that after we get this urgent fix in?